tests/unit/test_adapter.py

```diff
@@ -666,6 +674,77 @@ def test_expire_glue_table_versions(self, aws_credentials):
         # TODO moto issue https://github.com/getmoto/moto/issues/5952
         # assert len(result) == 3
 
+    @mock_athena
+    def test_get_work_group_output_location(self, aws_credentials):
+        self.adapter.acquire_connection("dummy")
+        self.mock_aws_service.create_work_group_with_output_location(ATHENA_WORKGROUP)
+        work_group_location = self.adapter.get_work_group_output_location()
+        assert work_group_location is not None
+
+    @mock_athena
+    def test_get_work_group_output_location_no_location(self, aws_credentials):
+        self.adapter.acquire_connection("dummy")
+        self.mock_aws_service.create_work_group_no_output_location(ATHENA_WORKGROUP)
+        work_group_location = self.adapter.get_work_group_output_location()
+        assert work_group_location is None
+
+    @pytest.fixture
+    def s3_client(self, aws_credentials):
+        with mock_s3():
+            conn = boto3.client("s3", region_name="us-east-1")
+            yield conn
+
+    @mock_s3
+    def test_upload_seed_to_s3(self, aws_credentials, s3_client):
+        seed_table = agate.Table.from_object(seed_data)
+        self.adapter.acquire_connection("dummy")
+
+        s3_client.create_bucket(Bucket=BUCKET)
+
+        location = self.adapter.upload_seed_to_s3(
+            s3_data_dir=f"s3://{BUCKET}",
+            s3_data_naming="schema_table",
+            external_location=None,
+            database_name="db_seeds",
+            table_name="data",
+            table=seed_table,
+        )
+
+        assert location == "s3://test-dbt-athena/db_seeds/data/"
+
+    @mock_s3
+    def test_upload_seed_to_s3_external_location(self, aws_credentials, s3_client):
+        seed_table = agate.Table.from_object(seed_data)
+        self.adapter.acquire_connection("dummy")
+
+        bucket = "my-external-location"
+        external_location = f"s3://{bucket}/seeds/one/"
+        s3_client.create_bucket(Bucket=bucket)
+
+        location = self.adapter.upload_seed_to_s3(
+            s3_data_dir=None,
+            s3_data_naming="schema_table",
+            external_location=external_location,
+            database_name="db_seeds",
+            table_name="data",
+            table=seed_table,
+        )
+
+        assert location == "s3://my-external-location/seeds/one/"
+
+    def test__parse_lf_tags(self):
```

Comment:
could you add some other tests to see if the lf tag creation works? in order to do so we might need to check in moto if we have the right mocking functions...

Reply:
I checked moto, unfortunately no way to mock the `lakeformation` client :(

Reply:
Ok, let's open an issue there to ask what we need, folks are pretty fast implementing new meatures.
They fix a bug that I requested last week and they are adding new supported glue functions.